service/host.go

```diff
@@ -115,32 +116,29 @@ func (uis *UIServer) modifyHost(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// determine what action needs to be taken
-	switch opts.Action {
-	case "updateStatus":
-		currentStatus := h.Status
-		newStatus := opts.Status
-		if !util.StringSliceContains(validUpdateToStatuses, newStatus) {
-			http.Error(w, fmt.Sprintf("'%v' is not a valid status", newStatus), http.StatusBadRequest)
-			return
-		}
+	currentStatus := h.Status
+	modifyResult, restErr := modifyHostStatus(h, opts, u)
 
-		if h.Provider == evergreen.ProviderNameStatic && newStatus == evergreen.HostDecommissioned {
-			http.Error(w, "cannot decommission static hosts", http.StatusBadRequest)
+	if restErr != nil {
+		switch restErr.Error() {
+		case fmt.Sprintf(UnrecognizedAction, opts.Action):
+			uis.WriteJSON(w, restErr.StatusCode, restErr.Message)
 			return
-		}
-
-		err := h.SetStatus(newStatus, u.Id, "")
-		if err != nil {
-			uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "Error updating host"))
+		default:
+			uis.LoggedError(w, r, restErr.StatusCode, restErr)
```

Comment:
I believe this solves [EVG-3012](https://jira.mongodb.org/browse/EVG-3102)

Reply:
Best way to verify this is to deploy the change to staging. If it's true, put both tickets in the title